src/NovelRT/Ecs/SystemScheduler.cpp

```diff
@@ -75,7 +75,9 @@ namespace NovelRT::Ecs
 
     void SystemScheduler::RegisterSystem(std::function<void(Timing::Timestamp, Catalogue)> systemUpdatePtr) noexcept
     {
-        Atom id = Atom::GetNextSystemId();
+        static AtomFactory& systemIdFactory = AtomFactoryDatabase::GetFactory("SystemId");
```

Comment:
How are these strings defined to determine which factory to get? Is this something that will just be predefined by the Engine? (Meaning we provide users with a specific set of strings for "pre-defined" factories else they manage their own?)

Reply:
If you check the implementation of GetFactory you'll see it just generates a new one for the string if it doesn't exist.

Reply:
Is there ever a case where an end user will want to use one of the internal factories?

Reply:
After discussion in Discord - generally speaking, these strings should not be used externally nor shared. May be refactored into a private header at some point, but for MVP this is alright.